tests/DotNet.Testcontainers.Tests/Unit/Builders/TestcontainerBuilderTest.cs

```diff
@@ -0,0 +1,68 @@
+namespace DotNet.Testcontainers.Tests.Unit.Builders;
+
+using DotNet.Testcontainers.Builders;
+using DotNet.Testcontainers.Configurations;
+using DotNet.Testcontainers.Containers;
+using Images;
+using Xunit;
+
+public class TestcontainerBuilderTest
+{
+  [Theory]
+  [InlineData("my.proxy.com", "bar", "my.proxy.com/bar:latest")]
+  [InlineData("my.proxy.com", "bar:latest", "my.proxy.com/bar:latest")]
+  [InlineData("my.proxy.com", "bar:1.0.0", "my.proxy.com/bar:1.0.0")]
+  [InlineData("my.proxy.com/my-path", "bar:1.0.0", "my.proxy.com/my-path/bar:1.0.0")]
+  [InlineData("my.proxy.com:443", "bar:1.0.0", "my.proxy.com:443/bar:1.0.0")]
+  [InlineData("my.proxy.com", "foo/bar:1.0.0", "my.proxy.com/foo/bar:1.0.0")]
+  [InlineData("my.proxy.com/my-path", "foo/bar:1.0.0", "my.proxy.com/my-path/foo/bar:1.0.0")]
+  [InlineData("my.proxy.com:443", "foo/bar:1.0.0", "my.proxy.com:443/foo/bar:1.0.0")]
+  [InlineData("my.proxy.com:443/my-path", "foo/bar:1.0.0", "my.proxy.com:443/my-path/foo/bar:1.0.0")]
+  [InlineData("my.proxy.com", "myregistry.azurecr.io/foo/bar:1.0.0", "myregistry.azurecr.io/foo/bar:1.0.0")]
+  [InlineData("my.proxy.com", "myregistry.azurecr.io:443/foo/bar:1.0.0", "myregistry.azurecr.io:443/foo/bar:1.0.0")]
+  public void ImageNameFromStringIsPrefixedWhenPrefixIsSet(string prefix, string originalImageName, string expectedName)
+  {
+    TestcontainersSettings.DockerHubImagePrefix = prefix;
```

Comment:
We just need to make sure, that the tests won't run with other tests. Changing the static property might not be a good idea with parallel tests.

Reply:
I wasn't aware tests are executed in parallel.

Fixed.

Reply:
Then maybe the xunit documentation needs fixing: https://xunit.net/docs/running-tests-in-parallel > "Custom Test Collections". Or am I missing something there?

Reply:
> That's not how it works :). You'll need a `CollectionDefinition` first and then set `DisableParallelization`. I can add it. See: [xunit/xunit#1999 (comment)](https://github.com/xunit/xunit/issues/1999#issuecomment-522635397).

Yep, saw that issue. From what I gather from the docs though that is not required. Maybe I'm wrong.

Reply:
Each test class is a unique test collection. By default, test collections run parallel. The unit tests in this library are in different test collections. Tests that access the static resource (almost all of them) can't run parallel.

> Other tests relying on `TestcontainersSettings.DockerHubImagePrefix` then just would have to be in the same collection, no?
> 
> Or am I missing something there?

Yes, exactly. Almost all tests rely on that static property. But putting all of them in the same collection is way too complicated. We can just take advantage of the `[CollectionDefinition(DisableParallelization = true)]` behavior:

> Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

Reply:
Ah, got it. Thanks!

Reply:
I added a simple `ParallelizationDisabled` collection. Guess that's ok for now.